### PR TITLE
feat(Button): added isLoading spinner

### PR DESCRIPTION
**Pull request by SantiMoncada**

### CODE REVIEW

File: src/Button/Button.tsx [hunk 1]
```diff
@@ -36,32 +38,41 @@ export const Button = ({
...
-          forwardedAs="span"
-        >
-          {children}
-        </StyledTypography>
-      )}
-      {endIcon && (
-        <Box ml={hasChildren ? 1 : undefined}>
-          <Icon icon={endIcon} size={sizeIcon[size]} />
-        </Box>
+      {isLoading && (
```
Comment (david-gomey) [inline src/Button/Button.tsx:47]: Perhaps we can add a ternary

```javascript
{isLoading ? (
          <Spinner size={SPINNER_VARIANT_MAPPING[size]} />
        ) : (
          <>
            {startIcon && (
              <Box mr={hasChildren ? 1 : undefined}>
                <Icon icon={startIcon} size={sizeIcon[size]} />
              </Box>
            )}
            {hasChildren && (
              <StyledTypography
                $color={color}
                $variant={variant}
                disabled={disabled}
                variant={TYPOGRAPHY_VARIANT_MAPPING[size]}
                forwardedAs="span"
              >
                {children}
              </StyledTypography>
            )}
            {endIcon && (
              <Box ml={hasChildren ? 1 : undefined}>
                <Icon icon={endIcon} size={sizeIcon[size]} />
              </Box>
            )}
          </>
        )}
```

File: src/Button/Button.tsx [hunk 2]
```diff
@@ -36,32 +38,41 @@ export const Button = ({
       $size={size}
       $variant={variant}
       $hasChildren={hasChildren}
+      $isLoading={isLoading}
```
Comment (david-gomey) [inline src/Button/Button.tsx:41]: Do you need to pass this prop?

File: src/Button/Button.tsx [hunk 3]
```diff
@@ -36,32 +38,41 @@ export const Button = ({
...
       disabled={disabled}
```
Comment (david-gomey) [inline src/Button/Button.tsx:42]: Perhaps Button should be disabled when isLoading?

**Review by david-gomey — CHANGES_REQUESTED** [review]

File: src/Button/Button.tsx (hunk 1 above)
Comment (SantiMoncada) [reply src/Button/Button.tsx:47]: If we don't render the content of the button and only the spinner, the button will change sizes.

Comment (david-gomey) [reply src/Button/Button.tsx:47]: Ok, we can review it when you want

File: src/Button/Button.tsx [hunk 4]
```diff
@@ -40,33 +42,41 @@ export const Button = React.forwardRef<HTMLElement, ButtonProps>(
...
+              <Icon icon={startIcon} size={sizeIcon[size]} />
+            </Box>
+          )}
+          {hasChildren && (
+            <StyledTypography
+              $variant={variant}
+              disabled={disabled}
+              variant={TYPOGRAPHY_VARIANT_MAPPING[size]}
+              forwardedAs="span"
+            >
```
Comment (EduardoRG) [inline src/Button/Button.tsx:70]: For this Typography, and looking at the Button spec from figma... are we missing the prop `fontWeight={700}` maybe?

**Review by EduardoRG — CHANGES_REQUESTED** [review]

**Pull request merged**